### PR TITLE
Fix merging of categoryExclusions for DoubleClick RTC

### PR DESCRIPTION
**Pull request by bradfrizzell**
CategoryExclusions was getting treated the same as targeting, but they are different. Targeting is an object, cE is an array. Need to merge separately.

### CODE REVIEW

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js [hunk 1]
```diff
@@ -646,19 +646,22 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
...
+          this.jsonTargeting_['targeting'] =
+              !!this.jsonTargeting_['targeting'] ?
+              deepMerge(this.jsonTargeting_['targeting'],
+                        rewrittenResponse) :
+              rewrittenResponse;
+        }
+        if (rtcResponse.response['categoryExclusions']) {
+          this.jsonTargeting_['categoryExclusions'] =
+              rtcResponse.response['categoryExclusions'].concat(
+                  this.jsonTargeting_['categoryExclusions'] || {});
```
Comment (glevitzky) [inline extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: If cE is an array, should this be `... || []`?

Comment (bradfrizzell) [reply extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: doh, fixed

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js [hunk 2]
```diff
@@ -646,19 +646,22 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
...
+              rtcResponse.response['targeting'],
+              rtcResponse.callout);
+          this.jsonTargeting_['targeting'] =
+              !!this.jsonTargeting_['targeting'] ?
+              deepMerge(this.jsonTargeting_['targeting'],
+                        rewrittenResponse) :
+              rewrittenResponse;
+        }
+        if (rtcResponse.response['categoryExclusions']) {
+          this.jsonTargeting_['categoryExclusions'] =
```
Comment (keithwrightbos) [inline extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: Let's dedupe

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js [hunk 3]
```diff
@@ -647,18 +648,26 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
...
+              rtcResponse.response['targeting'],
+              rtcResponse.callout);
+          this.jsonTargeting_['targeting'] =
+              !!this.jsonTargeting_['targeting'] ?
+              deepMerge(this.jsonTargeting_['targeting'],
+                  rewrittenResponse) :
+              rewrittenResponse;
+        }
+        if (rtcResponse.response['categoryExclusions']) {
+          [this.jsonTargeting_['categoryExclusions'],
```
Comment (keithwrightbos) [inline extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: Only need to initialize exclusions once so above do:
let exclusions;
...
if (rtcResponse.response['categoryExclusions']) {
   if (!exclusions) {
      exclusions = {};
      // populate it here
   }
   // iterate response cats
}
// Outside of rtc response array iteration
if (exclusions) {
    this.jsonTargeting_['cat...'] = Object.keys(exclusions);
}

**Pull request merged**